### PR TITLE
Allow http request by configuration

### PR DESCRIPTION
**Pull request by Zav**
EnforceHttps option has been added in the ClientCredentialsOptions.

### CODE REVIEW

File: src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenHttpMessageHandler.cs [hunk 1]
```diff
@@ -27,27 +28,29 @@ internal sealed class ClientCredentialsTokenHttpMessageHandler : PolicyHttpMessa
 
     private readonly IClientCredentialsTokenManagementService _tokenManagementService;
     private readonly string _clientName;
+    private readonly ClientCredentialsOptions _clientCredentialsOptions;
 
-    public ClientCredentialsTokenHttpMessageHandler(IClientCredentialsTokenManagementService tokenManagementService, string clientName)
+    public ClientCredentialsTokenHttpMessageHandler(IClientCredentialsTokenManagementService tokenManagementService, string clientName, ClientCredentialsOptions clientClientCredentialsOptions)
```
Comment (asimmon) [inline src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenHttpMessageHandler.cs:33]: In the [rest of the code](https://github.com/search?q=repo%3Agsoft-inc%2Fgsoft-authentication-clientcredentialsgrant+%22options+%3D+%22&type=code) we name the `ClientCredentialsOptions` instances `options` or `_options` so we might want to do this here too (both param + field)

File: src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsOptions.cs [hunk 2]
```diff
@@ -51,4 +51,9 @@ public string Scope
     /// Internal cache key computed by options post-configuration
     /// </summary>
     internal string CacheKey { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Enforce https for all authenticated requests
+    /// </summary>
+    public bool EnforceHttps { get; set; } = true;
```
Comment (asimmon) [inline src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsOptions.cs:58]: So do we want it to be enabled by default?

File: src/GSoft.Authentication.ClientCredentialsGrant.Tests/ClientCredentialsTokenHttpMessageHandlerTests.cs [hunk 3]
```diff
@@ -12,6 +12,7 @@ public sealed class ClientCredentialsTokenHttpMessageHandlerTests : IDisposable
     private readonly IClientCredentialsTokenManagementService _tokenManagementService;
     private readonly MockPrimaryHttpMessageHandler _mockPrimaryHttpMessageHandler;
     private readonly HttpClient _clientCredentialsHttpClient;
+    private readonly ClientCredentialsOptions _option;
```
Comment (asimmon) [inline src/GSoft.Authentication.ClientCredentialsGrant.Tests/ClientCredentialsTokenHttpMessageHandlerTests.cs:15]: `_options` (+s)

**Review by asimmon — DISMISSED** [review]

**Comment by Gcaya** [comment]
We might want write something about how you can enable http in the readme?

File: src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsOptions.cs (hunk 2 above)
Comment (Gcaya) [inline src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsOptions.cs:58]: Maybe we could say something about this option in the readme.

**Review by asimmon — APPROVED** [review]

**Pull request merged**